### PR TITLE
Automatically add add-on guid to manifest if none is given.

### LINKED ISSUE #11559: Insert add-on ID into manifest if it's missing
One of the issues that came up during Armagadd-on is that it's problematic for add-ons to not have an ID set in their manifest. A broken signature means there's no way for Firefox to figure out their ID.

AMO should add the [ID in the manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) if the uploaded add-on doesn't have it, before signing.

The reason we don't want to require developers to add the ID themselves is that we don't want to make it harder for them to maintain a Firefox and Chrome version of their add-on. In some cases it's possible to submit the same code for both browsers, so we should try to keep things that way if possible.

/cc @muffinresearch @ddurst

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #11559

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
We are still setting the `guid` on the `Addon` instance here: https://github.com/mozilla/addons-server/pull/11607/files#diff-d5d4f5e227c5a588706b9e9f6879197eR550 - that `data` comes from `parse_addon` which returns the GUID from the manifest.json instead then. Since this PR makes the guid be present in the manifest.json at any time that's how it works.

I'm working on fixing the tests.

File: src/olympia/files/tasks.py [hunk 1]
```diff
@@ -75,3 +78,36 @@ def repack_fileupload(upload_pk):
...
+
+    data = parse_xpi(upload.path, minimal=True)
+
+    generate_guid = (
+        not data.get('guid', None) and
+        data.get('is_webextension', False)
+    )
+
+    if generate_guid:
+        data['guid'] = guid = generate_addon_guid()
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/files/tasks.py]: Whoops, indeed. I think this is a relict from a copy&paste

**Comment by diox** [comment]
Ok, got it. Now, that would invalidate most of the conditions around `guid` presence in `signing/views.py` then, right ?

**Comment by EnTeQuAk** [comment]
> Ok, got it. Now, that would invalidate most of the conditions around guid presence in signing/views.py then, right ?

I don't think so. The main idea behind this is that Firefox is that we're able to double-check on AMO and Firefox or for debugging purposes that the signatures CN matches the add-on id.

The developer flow doesn't change by this I think. I don't think developers are pushing updated XPIs based on the signed ones we are returning to them - these are the only ones where we add the id in case it has been missing.

**Comment by diox** [comment]
But there are checks in `signing/views.py` that depend on the presence of a guid in the package. With your change, they are rendered completely irrelevant, since at that point the package *has* to have a guid...

**Comment by diox** [comment]
And actually I'm also worried about accidentally adding a random `guid` for updates as well... Previously we used to generate a `guid` only at add-on creation, ignoring that step for new versions. But since you're adding the guid at the upload step, wouldn't that force it at every upload, whether it's a new version or a new add-on entirely ?

**Comment by EnTeQuAk** [comment]
Closing for now, going to integrate mentioned changes and re-open as a draft PR later.

**Pull request closed without merging**